### ISSUE TITLE
Nan values in dewpoint_from_specific_humidity

### ISSUE THREAD

**Issue by OpheliaMiralles** [ecmwf/earthkit-meteo#20]
### What happened?

When the `t_from_es ` function gets called by `dewpoint_from_specific_humidity`, the np.log is called on a division where the denominator can be zero. It raises a silent error and produces nans in the result (es=0 and v is -inf).
If the specific humidity is zero, nans are systematically produced. 

However, shouldn't the dewpoint temperature from a specific humidity of 0.0 be defaulted to some meaningful value, representing the dewpoint temperature of dry air, very low, like 0K? It can be that the specific humidity is very close to 0 and due to rounding, datasets end up with nans, which can reveral rather painful when the data is intended to be used for training of ML models.

```
def t_from_es(self, es):
        v = np.log(es / self.c1)
        return (v * self.c4w - self.c3w * self.t0) / (v - self.c3w)
```

### What are the steps to reproduce the bug?

Call thermo.dewpoint_from_specific_humidity(q=y, p=x) with some 0 values in y and any pressure in Pa.

### Version

v0.6.2

### Platform (OS and architecture)

Linux balfrin-ln002 5.14.21-150400.24.81_12.0.87-cray_shasta_c

**Comment by sandorkertesz** [comment]
@OpheliaMiralles, thank you for reporting this issue. What do you suggest as the value for dewpoint temperature when the specific humidity is 0? I think it should be a user provided value.

**Comment by sandorkertesz** [comment]
@OpheliaMiralles, please see #23 providing a fix for it.

**Comment by OpheliaMiralles** [comment]
Many thanks, I think letting the user chose a value is the right way to do it indeed.

**Comment by sandorkertesz** [comment]
Thanks for the feedback. The fix has been released as version 0.2.0.

**Comment by sandorkertesz** [comment]
@OpheliaMiralles, upon further review of this change we decided it introduced too much complexity and had the potential to reduce performance. Therefore we have reverted this change and released version 0.3.0 without the extra parameters. 

The simplest solution for your case is just to replace nans with the required value in the results:

```python
from earthkit.meteo import thermo
import numpy as np

q = np.array([0.01, 0])
p = np.array([100000., 100000.])

td = thermo.dewpoint_from_specific_humidity(q, p)
np.nan_to_num(td, copy=False, nan=100.)
td
```
```
array([287.16144849, 100.        ])
```

Alternatively, if the input data can contain nans a mask can be used:

```python
from earthkit.meteo import thermo
import numpy as np

q = np.array([0.01, 0])
p = np.array([100000., 100000.])

mask = q < 1e-10

# prevent RuntimeWarning issued due to calling log with zero
q[mask] = np.nan

td = thermo.dewpoint_from_specific_humidity(q, p)

td[mask] = 100.
td
```
```
array([287.16144849, 100.        ])
```
